### PR TITLE
doc: add Bluetooth LE Controller documentation

### PR DESCRIPTION
**Pull request by ru-fu**
Add a user guide that describes what Bluetooth LE Controllers
are available and what they support.

Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_ble_controller.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+*******
+
+Most :ref:`Bluetooth LE samples <ble_samples>` in the |NCS| can use either Bluetooth LE Controller.
+An exception is the :ref:`ble_llpm` sample, which requires a controller that supports LLPM.
+Therefore, this sample must be run with the nRF Bluetooth LE Controller.
+
+By default, all samples except for the :ref:`ble_llpm` sample are currently configured to use Zephyr's Bluetooth LE Controller.
+To use the nRF Bluetooth LE Controller instead, complete the following steps:
+
+1. ...
```
Comment (grochu) [inline doc/nrf/ug_ble_controller.rst]: I would propose something like:
1. Run Zephyr interactive configuration tool: `ninja menuconfig`
2. Go to _"Bluetooth" -> "Blutooth Controller" -> "Blutooth Link Layer Selection"_
3. Select _"Nordic proprietary BLE Link Layer"_
4. Additionally set `CONFIG_BT_HCI_TX_STACK_SIZE` to 1536:
 - enable _"Bluetooth" -> "Override HCI Tx thread stack size"_
 - set _"Bluetooth" -> "HCI Tx thread stack size"_ to 1536
5. If Bluetooth: Central UART or Bluetooth: Central BAS samples are used, set `CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE` to 2048

@ru-fu please confirm steps 4. and 5. with @rugeGerritsen or @thomasstenersen 

Note that, the alternative path for this Kconfig Symbol in _menuconfig_ is:
_"Modules" -> "Nordic nrfxlib" -> "Bluetooth Link Layer Selection"_

Other alternative is to modify "prj.conf" file (in this case, user just needs to add line:
`CONFIG_BT_LL_NRFXLIB_DEFAULT=y` )

Comment (thomasstenersen) [reply doc/nrf/ug_ble_controller.rst]: I'm not sure if 4 and 5 are required. Would need to confirm again, but I don't have time to do that right now, unfortunately.

I would propose the latter alternative, to keep the instructions simple.

1. Ensure `CONFIG_BT_LL_NRFXLIB_DEFAULT=y` is set in the `prj.conf` file.
2. Build the example from a *clean* build directory.

**Comment by ru-fu** [comment]
Thanks for the input - I've updated the instructions.
PDF printout: 
[Bluetooth LE Controller — nRF Connect SDK 1.1.pdf](https://github.com/NordicPlayground/fw-nrfconnect-nrf/files/4139099/Bluetooth.LE.Controller.nRF.Connect.SDK.1.1.pdf)

File: doc/nrf/ug_ble_controller.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Zephyr Bluetooth LE Controller
+******************************
+
+The `Zephyr Project`_ provides a full :ref:`Bluetooth Low Energy stack <zephyr:bluetooth-arch>`, which includes a Bluetooth LE Controller.
+This Bluetooth LE Controller supports a wide variety of devices.
+It is developed by the Zephyr community and provided as open source.
+
+To use Zephyr's Bluetooth LE Controller in your application, include a :ref:`Controller-only build <zephyr:bluetooth-build-types>` of the Bluetooth LE stack.
+
+Zephyr's Bluetooth LE Controller supports most standard Bluetooth LE features, except for Advertising Extensions.
```
Comment (greg-fer) [inline doc/nrf/ug_ble_controller.rst]:
```suggestion
Zephyr's Bluetooth LE Controller supports most of the standard Bluetooth LE features, except for Advertising Extensions.
```

File: doc/nrf/ug_ble_controller.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
+.. _ug_ble_controller:
+
+Bluetooth LE Controller
+#######################
+
+When you develop a Bluetooth Low Energy (LE) application, you must include a Bluetooth LE Controller.
+A Bluetooth LE Controller is the layer of the Bluetooth stack that handles the physical layer packets and all associated timing.
+It implements the Link Layer, which is the low-level, real-time protocol that controls the Bluetooth LE communication.
+
+The |NCS| provides different implementations of a Bluetooth LE Controller.
```
[the remark this answers is not in the document]
Comment (ru-fu) [reply doc/nrf/ug_ble_controller.rst]: Yes, since we include a Zephyr fork in the NCS, the NCS provides it.
I'll add the links.

**Comment by ru-fu** [comment]
Ready to merge when CI completes.

**Pull request merged**